python/ray/serve/deployment_state.py

```diff
@@ -202,7 +203,9 @@ def __init__(
         # Populated in self.stop().
         self._graceful_shutdown_ref: ObjectRef = None
 
+        # todo: will be confused with deployment_config.is_cross_language
         self._is_cross_language = False
+        self._deployment_is_cross_language = False
```

Comment:
What's the difference between them?

Reply:
`self._is_cross_language` indicates that the language of the deployment is not python. `self._deployment_is_cross_language` represents the `is_cross_language` parameter passed when the deployment was created. This parameter is configured based on the language of the client called at that time.

Reply:
So maybe we should name them to:
_is_cross_language -> deployment_language
_deployment_is_cross_language -> _deployed_from_java

Reply:
We have four scenarios:
![image](https://user-images.githubusercontent.com/11265783/182834710-c32c3900-9216-4078-a75d-c9c11acc04e3.png)
If we add c++, go serve in the future, it will become unmaintainable. 
That's another big issue. We need to solve how deployments in different languages are created and managed by clients in different languages. For example, a python client creates a python deployment. And we use a java client to manage this deployment. But currently, it will return some pickled data to the java clients.

Reply:
OK I see. Thanks for the declaration.